File: Java/Base64Encoding.java
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.Base64;

// Basic authentication example
public class Base64Encoding {

    public static void main(String[] args) throws IOException, InterruptedException {

        // Specify customer key
        final String customerKey = "Your customer key";
        // Specify customer ID
        final String customerSecret = "Your customer secret";

        // Concatenate and encode with base64
        String plainCredentials = customerKey + ":" + customerSecret;
        String base64Credentials = new String(Base64.getEncoder().encode(plainCredentials.getBytes()));
        // Create authorization header
        String authorizationHeader = "Basic " + base64Credentials;

        HttpClient client = HttpClient.newHttpClient();

        String body = "{\"destination\": \"userB\", \"enable_offline_messaging\": false,  \"enable_historical_messaging\": false,    \"payload\": \"Hello\"}";

        // Build HTTP request
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("https://api.agora.io/dev/v2/project/<Your App ID>/rtm/users/userA/peer_messages?wait_for_ack=true"))
                .POST(HttpRequest.BodyPublishers.ofString(body))
                .header("Authorization", authorizationHeader)
                .header("Content-Type", "text/plain")
                .build();
        // Send HTTP request
        HttpResponse<String> response = client.send(request,
                HttpResponse.BodyHandlers.ofString());

        System.out.println(response.body());
    }
}

File: PHP/base64encoding.php
<?php
// Basic authentication example
// Your customer key
$customerKey = "Your customer key";
// Your customer secret
$customerSecret = "Your customer secret";
// Create a string: customerKey:customerSecret
$credentials = $customerKey . ":" . $customerSecret;

// Use base64 encoding
$base64Credentials = base64_encode($credentials);
// Create authorization header
$arr_header = "Authorization: Basic " . $base64Credentials;

$curl = curl_init();
// Send request with specified header and body
curl_setopt_array($curl, array(
  CURLOPT_URL => 'https://api.agora.io/dev/v2/project/<Your App ID>/rtm/users/userA/peer_messages?wait_for_ack=true',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'POST',
  CURLOPT_POSTFIELDS =>'{
    "destination": "userB",
    "enable_offline_messaging": false,
    "enable_historical_messaging": false,
    "payload": "Hello"
}',
  CURLOPT_HTTPHEADER => array(
    $arr_header,
    'Content-Type: text/plain'
  ),
));

$response = curl_exec($curl);

if($response === false) {
    echo "Error in cURL : " . curl_error($curl);
}

curl_close($curl);

echo $response;